proposals/003-metric-improvenents.md

```diff
@@ -0,0 +1,262 @@
+| kroxylicious_downstream_backpressure_time        | Distribution | virtual_cluster, node_id†                                                             | Records the time that reading from the downstream has been paused owing to backpressure |
+| kroxylicious_upstream_backpressure_state         | Gauge        | virtual_cluster, node_id†                                                             | Records the backpressure state from the upstream                                        |
+| kroxylicious_upstream_backpressure_time          | Distribution | virtual_cluster, node_id†                                                             | Records the time that reading from the upstream has been paused owing to backpressure   |
+
+The following metrics wil have changes to their labels.
+
+| Metric                                        | Type    | Labels                     | Deprecated Labels | Description                                                                         |
+|-----------------------------------------------|---------|----------------------------|-------------------|-------------------------------------------------------------------------------------|
+| kroxylicious_downstream_connections_total     | Counter | virtual_cluster, node_id†  | virtualCluster    | Incremented by one every time a request comes from/response goes to the downstream. |
+| kroxylicious_downstream_error_total           | Counter | virtual_cluster†, node_id† | virtualCluster    | Incremented by one every time a request goes to/response comes from the upstream.   |
+| kroxylicious_upstreamstream_connections_total | Counter | virtual_cluster,  node_id† | virtualCluster    | Incremented by one every time a request comes from/response goes to the downstream. |
+| kroxylicious_upstreamstream_error_total       | Counter | virtual_cluster,  node_id† | virtualCluster    | Records the time taken for the message to traverse the proxy                        |
+
+
+The following metrics will be deprecated.
+
+| Metric                                           |
+|--------------------------------------------------|
+| kroxylicious_inbound_downstream_messages         |
+| kroxylicious_inbound_downstream_decoded_messages |
+| kroxylicious_payload_size_bytes                  |
+
+
+(† signifies labels that will be omitted in some circumstances. The circumstances are explained below).
+
+### New counter `kroxylicious_downstream_kafka_message_total`
+
+`kroxylicious_downstream_kafka_message_total` is incremented every time a message passes to/from the downstream.
+It is discriminated by `virtual_cluster`, `node_id`, `flowing`, `opaque`, api_key`, and `api_version` labels.
+
+`node_id` will only be present if for requests going to or responses coming from a broker.  For bootstrap interactions,
+`node_id` will be absent.
```

Comment:
It being absent is not very self-describing. Perhaps we could use `node_id=bootstrap` in this case.